src/worker/jobs/expiry_notification.rs

```diff
@@ -41,31 +45,31 @@ impl BackgroundJob for CheckAboutToExpireToken {
 fn check(emails: &Emails, conn: &mut PgConnection) -> anyhow::Result<()> {
     info!("Checking if tokens are about to expire");
     let expired_tokens = find_tokens_expiring_within_days(conn, EXPIRY_THRESHOLD)?;
-    // Batch send notifications in transactions.
-    const BATCH_SIZE: usize = 100;
-    for chunk in expired_tokens.chunks(BATCH_SIZE) {
+    if expired_tokens.len() == MAX_ROWS as usize {
+        warn!("The number of tokens to check exceeds the limit");
+    }
```

Comment:
I'm not sure if warning about `expired_tokens.len() == MAX_ROWS` **exceeding the limit** is a good idea. Since it's an equality check. A better approach might be to compare with `expired_tokens.len() > MAX_ROWS`. We could then query for `MAX_ROWS + 1` elements and process only the first `MAX_ROWS`.

Reply:
hmm, IMHO it's fine as is. though we could potentially rephrase the warning message a bit to make it clearer.

```suggestion
    if expired_tokens.len() == MAX_ROWS as usize {
        warn!("The max. number of API tokens per query has been reached. More API tokens might be processed on the next run.");
    }
```

Reply:
Yes, I agree that we need to improve the warning message. `Exceeds` is quite confusing.